ui/js/controllers/perf/compare.js

```diff
@@ -244,61 +297,98 @@ perf.controller('CompareResultsCtrl', [
-                                    newSeriesList,
-                                    [$scope.newResultSet.id]);
-                            }).then(function (resultMaps) {
-                                var newResultsMap = resultMaps[$scope.newResultSet.id];
+                PhSeries.getSeriesList(
+                    $scope.originalProject.name,
+                    {interval: timeRange, subtests: 0,
+                        framework: $scope.filterOptions.framework.id
+                    }).then(
+                        function(originalSeriesList) {
+                            $scope.platformList = _.uniq(
+                                _.map(originalSeriesList, 'platform'));
+                            $scope.testList = _.uniq(
+                                _.map(originalSeriesList, 'name'));
+                            return PhCompare.getResultsMap($scope.originalProject.name,
+                                                        originalSeriesList,
+                                                        {pushIDs: resultSetIds});
+                        }).then(function(resultMaps) {
+                            var originalResultsMap = resultMaps[$scope.originalResultSet.id];
+                            var newResultsMap = resultMaps[$scope.newResultSet.id];
+
+                            // Optimization - we collected all data in a single pass
+                            if (newResultsMap) {
                                 $scope.dataLoading = false;
                                 displayResults(originalResultsMap, newResultsMap);
-                            });
-                    });
-        }
+                                return;
+                            }
 
+                            PhSeries.getSeriesList(
+                                $scope.newProject.name,
+                                { interval: timeRange, subtests: 0,
+                                    framework: $scope.filterOptions.framework.id }).then(
+                                    function(newSeriesList) {
+                                        $scope.platformList = _.union(
+                                            $scope.platformList,
+                                            _.uniq(_.map(newSeriesList, 'platform')));
+                                        $scope.testList = _.union(
+                                            $scope.testList,
+                                            _.uniq(_.map(newSeriesList, 'name')));
+
+                                        return PhCompare.getResultsMap($scope.newProject.name,
+                                                                    newSeriesList,
+                                                                    {pushIDs: [$scope.newResultSet.id]});
+                                    }).then(function(resultMaps) {
+                                        var newResultsMap = resultMaps[$scope.newResultSet.id];
```

Comment:
No need for the `newResultsMap` variable, just pass `resultMaps[$scope.newResultSet.id]` to the function directly